### ISSUE TITLE
Only prefix commands show up in help menu

### ISSUE THREAD

**Issue by vicky5124** [kangalioo/poise#13]
If a context menu command is registered, it will not show up in help, even if it's also allowed to run as a slash or prefix command, though, that should probably not matter and show up anyway.

```rust
/// Test
#[poise::command(context_menu_command = "Test", slash_command, prefix_command)]
pub async fn test(
    ctx: Context<'_>,
    #[description = "test"] _user: User,
) -> CommandResult {
    Ok(())
}
```
```
Commands:
  /help        Show this menu
  
Test bottom text
```

Running `/help test` doesn't work either, saying "No such command `test`"

**Comment by vicky5124** [comment]
This issue is not over, context menu commands don't show up.
And as an additional issue, dynamic prefixes don't show in help either.

![00 58 53_05-12-2021](https://user-images.githubusercontent.com/8344197/144728300-6cff2c93-23ab-4087-8d6f-a2865c532b5b.png)

And I have a feeling that over 1 in length prefixes will not align the description text.

**Comment by kangalioo** [comment]
Technically, these are three separate issues :smile: I'll see if I can fix them quickly

**Comment by kangalioo** [comment]
What do you think of this output for context menu commands?
```
Commands:
  /help        Show this help menu
  /vote        Vote for something
  /getvotes    Retrieve number of votes
  /add         Add two numbers
  /choice      Dummy command to test slash command choice parameters
  /boop        Boop the bot!
  /delete      Deletes the given message
  /user_info   Query information about a Discord profile
  /echo        Echo content of a message
  /greet       Greet a user. Showcasing autocomplete!

Context menu commands:
  User information (on user)
  Echo (on message)

This is an example bot made to showcase features of my custom Discord bot framework
```

**Comment by vicky5124** [comment]
Great! But, how will it fair with command descriptions? How does `/help command name` get affected by this? Will it be case-insensitive, or will the user have to write the command name entirely as is?

**Comment by kangalioo** [comment]
Good point, thanks for noticing that. I made it case-insensitive, and you can refer to commands by either their prefix, slash, or context menu name
![Screenshot_20211206_215556](https://user-images.githubusercontent.com/21220820/144921288-461a791f-3f7b-44cd-8ecd-41ec50911c17.png)

**Comment by kangalioo** [comment]
Implemented in f533fdd2fd77aee4a7b1470d5fd2fe448711b6fb d016c7873a464dc452e5448dbf8eca773027e2ad 90a1ef00a214a60a758d0a57b33c3d08c0d2cab2 c90727cd39c468ac3b343b7fb36f694d5eab8ea9
